core/src/saros/misc/xstream/XStreamFactory.java

```diff
@@ -0,0 +1,62 @@
+package saros.misc.xstream;
+
+import com.thoughtworks.xstream.XStream;
+import com.thoughtworks.xstream.io.HierarchicalStreamDriver;
+import com.thoughtworks.xstream.security.NoTypePermission;
+import com.thoughtworks.xstream.security.NullPermission;
+import com.thoughtworks.xstream.security.PrimitiveTypePermission;
+import java.util.Collection;
+import java.util.Map;
+
+/** Factory class providing XStream objects with the correctly set-up security framework. */
+public class XStreamFactory {
+
+  /**
+   * Returns an <code>XStream</code> object with an set-up security framework.
+   *
+   * @return an <code>XStream</code> object with an set-up security framework
+   */
+  public static XStream getSecureXStream() {
+    XStream xStream = new XStream();
+
+    setUpSecurityFramework(xStream);
+
+    return xStream;
+  }
+
+  /**
+   * Returns an <code>XStream</code> object with an set-up security framework. The passed <code>
+   * HierarchicalStreamDriver</code> will be used when instantiating the <code>XStream</code> object
+   *
+   * @return an <code>XStream</code> object with an set-up security framework
+   */
+  public static XStream getSecureXStream(HierarchicalStreamDriver hierarchicalStreamDriver) {
+    XStream xStream = new XStream(hierarchicalStreamDriver);
+
+    setUpSecurityFramework(xStream);
+
+    return xStream;
+  }
+
+  /**
+   * Sets up the security framework for the passed <code>XStream</code> object.
+   *
+   * @param xStream the <code>XStream</code> object to set the security framework up for
+   * @see <a
+   *     href="https://x-stream.github.io/security.html">https://x-stream.github.io/security.html</a>
+   */
+  private static void setUpSecurityFramework(XStream xStream) {
+    // forbid all classes by default
+    xStream.addPermission(NoTypePermission.NONE);
+
+    // allow default java stuff
+    xStream.addPermission(NullPermission.NULL);
+    xStream.addPermission(PrimitiveTypePermission.PRIMITIVES);
+    xStream.allowTypeHierarchy(Collection.class);
+    xStream.allowTypeHierarchy(Map.class);
+    xStream.allowTypes(new Class[] {String.class});
+
+    // allow all saros classes
+    xStream.allowTypesByWildcard(new String[] {"saros.**"});
```

Comment:
In my opinion we should reduce this further to allow only primitive types or special classes that are only containing primitive data

Reply:
I think already implemented it that way at first by going through all fields that were annotated with an xsteam alias, but this way seemed cleaner. Especially since it does not require constant updates if you add new field types.

But I will push the other version later today (probably around 1 pm) so that you can have a look at it and compare the two.

Reply:
As far as I can tell, this is the minimal setup. Furthermore, it requires additional setup in `SPathConverterTest`. The amended PR is available on https://github.com/saros-project/saros/tree/alternative_xstream_security.

```suggestion
    // allow classes needed for Saros network logic
    xStream.allowTypeHierarchy(IActivity.class);
    xStream.allowTypeHierarchy(SarosPacketExtension.class);
    xStream.allowTypeHierarchy(Operation.class);

    Class[] allowedSarosNetworkClasses =
        new Class[] {
          FileList.class,
          JID.class,
          JupiterVectorTime.class,
          ProjectNegotiationData.class,
          SPath.class,
          User.class,
          UserListExtension.UserListEntry.class,
          VersionExchangeExtension.class,
          XStreamPacketExtension.class,
        };
    xStream.allowTypes(allowedSarosNetworkClasses);

    String[] allowedPrivateSarosNetworkClassNames = {
      FileList.class.getCanonicalName() + "$File", FileList.class.getCanonicalName() + "$MetaData"
    };
    xStream.allowTypes(allowedPrivateSarosNetworkClassNames);

    // allow classes needed for ColorIDSetStorage
    Class[] allowedColorIDSetStorageClasses = new Class[] {ColorIDSet.class, UserColorID.class};
    xStream.allowTypes(allowedColorIDSetStorageClasses);

    // allow classed needed for XMPPAccountStore
    Class[] allowedXMPPAccountStoreClasses = new Class[] {XMPPAccount.class};
    xStream.allowTypes(allowedXMPPAccountStoreClasses);

    String[] allowedPrivateXMPPAccountStoreClassName = {
      XMPPAccountStore.class.getCanonicalName() + "$AccountStoreInformation"
    };
    xStream.allowTypes(allowedPrivateXMPPAccountStoreClassName);
```